### PR TITLE
build: Bump go to version 1.22.2

### PR DESCRIPTION
**Pull request by brianmcarey**
### What this PR does

Kubernetes is building with go version 1.22[1]

KubeVirt should align with this.

go v1.22.2 includes a security fix for net/http and some more bug
fixes[2]

Build KubeVirt with go v1.22.2

[1] https://github.com/kubernetes/kubernetes/blob/b3926d137cd2964cd3a04088ded30845910547b1/go.mod#L9
[2] https://github.com/golang/go/issues?q=milestone%3AGo1.22.2+label%3ACherryPickApproved

Fixes #

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: 

### Special notes for your reviewer

Majority of changes come from `make generate` changing the formatting of descriptions - https://github.com/kubevirt/kubevirt/pull/11482/commits/8bf4abfb1e8c184011c70ab583a17708cbd98db9

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
Build KubeVirt with go v1.22.2
```

### CODE REVIEW

**Review by dhiller — COMMENTED** [review]
/approve

**Review by EdDev — APPROVED** [review]
Thank you!

**Review by xpivarc — COMMENTED** [review]
/approve
/hold

File: hack/builder/Dockerfile [hunk 1]
```diff
@@ -86,7 +86,7 @@ RUN set -x && \
     go install -v github.com/golang/protobuf/protoc-gen-go@1643683 && \
     go install -v k8s.io/code-generator/cmd/client-gen@v0.20.2 && \
     go install -v github.com/securego/gosec/v2/cmd/gosec@0ce48a5 && \
-    go install -v sigs.k8s.io/controller-tools/cmd/controller-gen@v0.4.0 && \
+    go install -v sigs.k8s.io/controller-tools/cmd/controller-gen@v0.14.0 && \
```
[the remark this answers is not in the document]
Comment (brianmcarey) [reply hack/builder/Dockerfile:89]:
```
+ controller-gen crd:allowDangerousTypes=true paths=../api/core/v1/
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x9d55af]

goroutine 15 [running]:
go/types.(*Checker).handleBailout(0xc00059fc00, 0xc000ce3d40)
	/gimme/.gimme/versions/go1.22.2.linux.amd64/src/go/types/check.go:367 +0x88
panic({0xb8f160?, 0x1246800?})
```

File: WORKSPACE [hunk 2]
```diff
@@ -44,11 +44,10 @@ rules_proto_toolchains()
 
 http_archive(
     name = "io_bazel_rules_go",
-    sha256 = "91585017debb61982f7054c9688857a2ad1fd823fc3f9cb05048b0025c47d023",
+    sha256 = "80a98277ad1311dacd837f9b16db62887702e9f1d1c4c9f796d0121a46c8e184",
     urls = [
-        "https://mirror.bazel.build/github.com/bazelbuild/rules_go/releases/download/v0.42.0/rules_go-v0.42.0.zip",
-        "https://github.com/bazelbuild/rules_go/releases/download/v0.42.0/rules_go-v0.42.0.zip",
-        "https://storage.googleapis.com/builddeps/91585017debb61982f7054c9688857a2ad1fd823fc3f9cb05048b0025c47d023",
+        "https://mirror.bazel.build/github.com/bazelbuild/rules_go/releases/download/v0.46.0/rules_go-v0.46.0.zip",
```
Comment (akalenyu) [inline WORKSPACE:49]: Yeah and specifically this https://github.com/bazelbuild/rules_go/pull/3756

File: hack/builder/Dockerfile (hunk 1 above)
[the remark this answers is not in the document]
Comment (xpivarc) [reply hack/builder/Dockerfile:89]: Thanks

[the remark this answers is not in the document]
Comment (akalenyu) [reply hack/builder/Dockerfile:89]: https://github.com/kubernetes-sigs/controller-tools/issues/888

**Pull request merged**